### PR TITLE
Enhance EnsureRecordTypes to work properly with Case, Lead, and Solution

### PR DESCRIPTION
**Pull request by davidmreed**
# Critical Changes

# Changes

- The `EnsureRecordTypes` class is now exposed as `ensure_record_types` and correctly supports the Case, Lead, and Solution sObjects (in addition to other standard objects).

# Issues Closed

### CODE REVIEW

File: cumulusci/tasks/salesforce/EnsureRecordTypes.py [hunk 1]
```diff
@@ -116,11 +120,19 @@ def _build_package(self):
                         self.options["sobject"]
                     )
                 )
+                # Lead, Case, and Solution require a default value in their Business Process
+                # Opportunity prohibits it.
+                if self.options["sobject"] == "Opportunity":
+                    default = "false"
+                else:
+                    default = "true"
```
Comment (jstvz) [inline cumulusci/tasks/salesforce/EnsureRecordTypes.py]: 💭 :
```python
       default = self.options["sobject"] != "Opportunity"
```

Then again, the existing version is probably more readable.

File: cumulusci/cumulusci.yml [hunk 2]
```diff
@@ -98,6 +98,13 @@ tasks:
         options:
             command: 'force:source:push'
         group: Salesforce DX
+    ensure_record_types:
```
Comment (jstvz) [inline cumulusci/cumulusci.yml]: 🤔 : Why did I think this was already a task?

Comment (davidmreed) [reply cumulusci/cumulusci.yml]: It was, but it wasn't exposed in a defined task - just per-project.

File: cumulusci/tasks/salesforce/EnsureRecordTypes.py (hunk 1 above)
Comment (davidmreed) [reply cumulusci/tasks/salesforce/EnsureRecordTypes.py]: Yeah, I was expecting more branches here and then discovered Case, Lead, and Solution all play by the same rules, but Opportunity doesn't. Adopting your proposal.

Comment (davidmreed) [reply cumulusci/tasks/salesforce/EnsureRecordTypes.py]: (Not before I make a couple of bone-headed mistakes, of course)

**Pull request merged**